os-server/src/main/java/app/openschool/auth/AuthServiceImpl.java

```diff
@@ -33,8 +54,30 @@ public User register(UserRegistrationDto userDto) {
       throw new EmailAlreadyExistException();
     }
 
-    User user = UserRegistrationMapper.userRegistrationDtoToUser(userDto, passwordEncoder);
-    return userRepository.save(user);
+    User user =
+        userRepository.save(
+            UserRegistrationMapper.userRegistrationDtoToUser(userDto, passwordEncoder));
+    sendEmailToVerifyUserAccount(user);
```

Comment:
we should do it asynchronously take a look on @Async

Reply:
move the logic about email sending into the separate service CommunicationService with EmailService implementation